### ISSUE TITLE
SDC burning timestep violation checks

### ISSUE THREAD

**Issue by zingale** [AMReX-Astro/Castro#856]
Currently we check for a burning timestep violation during each simplified SDC iteration, potentially triggering a retry.  But the way SDC works, the burning may become better behaved in latter iterations, and what may be a violation in iteration 1 could be fine in the last iteration once advection and burning have come to terms with each other.   Therefore, we should only trigger a retry on the last iteration if we can't meet the burning constraints.

**Comment by maxpkatz** [comment]
While I like the idea, the problem is that when the burn was unsuccessful, it often means that the integration failed in the middle of the integration and the state we get back is not a well-posed one. So trying to use it for the next SDC iteration could result in very bad things. This proposal might work if we could distinguish the failures where we finished the burn but didn't hit the tolerances from the failures where we never finished the burn. However I think if we did #883 that would largely supersede the need for this.
